tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -384,9 +384,10 @@ Status StorageManager::array_consolidate(
   }
 
   // Consolidate
-  Consolidator consolidator(this);
-  return consolidator.consolidate(
-      array_name, encryption_type, encryption_key, key_length, config);
+  auto mode = Consolidator::mode_from_config(config);
```

Comment:
This looks like it could be encapsulated in the factory function `Consolidator::create`. Is there are a reason we can't do that? In other words, make `mode_from_config` private, and use it to check and set the mode directly based on the config inside of the factory.

Reply:
Yes one call to create don't use the config but the mode directly. Ie: auto consolidator =
      Consolidator::create(ConsolidationMode::ARRAY_META, config, this);

Reply:
Marking resolved.